app.rb

```diff
@@ -1,136 +1,112 @@
 require_relative 'book'
-require_relative 'person'
+require_relative 'rental'
 require_relative 'student'
 require_relative 'teacher'
+require_relative 'classroom'
 
 # create app class
 class App
+  attr_accessor :rentals, :books, :people
+
   def initialize
-    @people = []
-    @books = []
     @rentals = []
+    @books = []
+    @people = []
   end
 
   def list_books
-    @books.each do |book|
-      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    if @books.empty?
+      puts 'No Books'
+    else
+      @books.each_with_index do |book, i|
+        puts "Number: #{i + 1}, Title: #{book.title}, Author: #{book.author}"
+      end
     end
   end
 
   def list_people
-    @people.each do |person|
-      puts "[#{person.class.name}] Name:#{person.name} ID:#{person.id} Age:#{person.age}"
+    if @people.empty?
+      puts 'No People'
+    else
+      @people.each_with_index do |person, i|
+        puts "No: #{i + 1}, Name: #{person.name},age: #{person.age} ID: #{person.id}"
+      end
     end
   end
 
   def create_person
-    puts "\nCreate a person (1) Student or (2) Teacher?"
-    option = gets.chomp.to_i
-
-    print 'Age:'
+    puts "\n"
+    puts 'Do you want to create a student (1) or teacher (2)?:'
+    type = gets.chomp.to_i
+    puts 'Enter person age:'
     age = gets.chomp.to_i
-    print 'Name:'
+    puts 'Enter person name:'
     name = gets.chomp
-
-    case option
-    when 1
-      create_student(age, name)
-    when 2
-      create_teacher(age, name)
+    if type == 1
+      puts 'Has parent permission? [Yes/No]'
+      has_parent_permission = gets.chomp
+      create_student(age, name, has_parent_permission)
+    elsif type == 2
+      puts 'Enter specialization'
+      specialization = gets.chomp
+      create_teacher(age, specialization, name)
     else
-      puts 'Invalid option.'
+      puts 'Invalid input'
     end
   end
```

Comment:
- [ ] As stated earlier,  Your App class is responsible for both user interaction (getting input from the user) and managing the data (books, people, rentals). Ideally, the responsibility for user interaction should be separated from the data management logic. This follows the Single Responsibility Principle and makes the code more modular and maintainable. To achieve this, you can create separate classes or modules for user interaction (getting input and displaying messages).